torch/testing/_internal/common_nn.py

```diff
@@ -3796,6 +3796,13 @@ def cross_entropy_loss_reference(input, target, weight=None, ignore_index=-100,
         reduction=reduction)
 
 
+def cross_entropy_loss_with_soft_labels_reference(input, target, reduction='mean'):
+    return kldivloss_reference(
+        torch.log_softmax(input, 1),
+        target,
+        reduction=reduction)
+
```

Comment:
Somewhere we should test that (1) cross_entropy_loss_with_soft_labels requires its input to be 2D, (2) cross_entropy_loss_with_soft_labels requires the input and target to be of the same shape and (3) the error messages for (1) and (2) are sane

Reply:
Agreed, I'm going to add a set of tests into `test_nn.py` for now. Longer term, this sort of thing will likely be covered by tests generated from a `ModuleInfo` entry.

Reply:
Agreed, the error cases will be covered in new tests within `test_nn.py`